pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+        # TODO: Implement reuse with new uri style
+        alias = uuid4().hex
+        try:
+            connections.connect(alias=alias, uri = self.uri)
+            logger.debug("Created new connection using: %s", alias)
+            return alias
+        except MilvusException as ex:
+            logger.error("Failed to create new connection using: %s", alias)
+            raise ex
+
+    def _init(self, input_data: Optional[dict]):
+        """Create/connect to the colletion"""
+        # If no input data and collection exists, use that
+        if input_data is None and utility.has_collection(
+            self.collection_name, using=self.alias
+        ):
+            self.collection = Collection(self.collection_name, using=self.alias)
+            # Grab the field information from the existing collection
+            self._extract_fields()
+        # If data is supplied we can create a new collection
+        elif input_data is not None:
+            self._create_collection(input_data)
+        # Nothin to init from
+        else:
+            logger.debug(
+                "No information to perform init from for collection %s",
+                self.collection_name,
+            )
+            return
+        self._create_index()
+        # Partitions only allowed on Milvus at the mmoment
+        if self.is_self_hosted and self.partitions is not None:
+            self.add_partitions(self.partitions)
+        self._create_default_search_params()
+        self._load()
+
+    def _create_collection(self, data: dict) -> None:
+        """Create the collection by autoinferring the schema."""
+        # TODO: Assuming ordered dict for 3.7
+        fields = {}
+
+        # Figure out each datatype of the input.
+        for key, value in data.items():
+            # Infer the corresponding datatype of the metadata
+            dtype = infer_dtype_bydata(value)
+
+            # Datatype isnt compatible
+            if dtype in (DataType.UNKNOWN, DataType.NONE):
+                logger.error(
+                    "Failed to parse schema for collection %s, unrecognized dtype for key: %s",
+                    self.collection_name,
+                    key,
+                )
+                raise ValueError(f"Unrecognized datatype for {key}.")
+
+            # Create an entry under the field name
+            fields[key] = {}
+            fields[key]["name"] = key
+            fields[key]["dtype"] = dtype
+
+            # Area for attaching kwargs for certain datatypes
+            if dtype == DataType.VARCHAR:
+                fields[key]["max_length"] = 65_535
+
+        if self.vector_field is None:
+            logger.error(
+                "Missing vector_field, cannot infer schema for collection: %s",
+                self.collection_name,
+            )
+            raise ValueError("Missing vector_field, cannot autoinfer schema.")
+
+        try:
+            self.dim = len(data[self.vector_field])
```

Comment:
The dimension calculation is correct here for float vectors, but wrong for binary vectors. In fact, we already got the field type by `infer_dtype_bydata`, which can help us to determine the dimension calculation more clear.

Reply:
Btw, `self.vector_field_type` should be also added. Or should we forbid the binary vector type?

Reply:
we should support for binary vector, but by default we assume it's float vector

Reply:
pymilvus checks for bytestrings so thats what I did here. If its binary vector the dim is 8 * the bystring length.